lib/internal/bootstrap/pre_execution.js

```diff
@@ -158,6 +165,31 @@ function setupSignalHandlers() {
   process.on('removeListener', stopListeningIfSignal);
 }
 
+
+function setupStacktracePrinterOnSigint() {
+  const { SigintWatchdog } = require('internal/watchdog');
+  const watchdog = new SigintWatchdog();
+  watchdog.oninterrupt = hideStackFrames(function interrupt() {
+    if (process.listenerCount('SIGINT') > 0) {
+      // Simply ignore this watchdog interrupt to prevent duplicate
+      // handling of SIGINT.
+      return;
+    }
+    if (process.stdout.isTTY) {
```

Comment:
I think if we print using `console.error()` (which I agree with is correct, we should be checking stderr here

```suggestion
    if (process.stderr.isTTY) {
```

(Although we could maybe also move this check to the beginning of `setupStacktracePrinterOnSigint()` and skip the setup altogether in that scenario?)